src/Stratis.FederatedPeg.Features.FederationGateway/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1083,7 +1088,10 @@ private void PutTransfers(DBreeze.Transactions.Transaction dbreezeTransaction, I
 
             // Write each transfer in order.
             foreach (ICrossChainTransfer transfer in orderedTransfers)
-                dbreezeTransaction.Insert(transferTableName, transfer.DepositTransactionId.ToBytes(), transfer);
+            {
+                byte[] transferBytes = this.dBreezeSerializer.Serialize(transfer);
```

Comment:
`this.dbreezeSerializer` is not required if doing it this way:

`byte[] crossChainTransferBytes = crossChainTransfer.ToBytes(this.network.Consensus.ConsensusFactory);`